pdns/calidns.cc

```diff
@@ -316,14 +318,28 @@ try
     vector<uint8_t> packet;
     DNSPacketWriter::optvect_t ednsOptions;
     boost::trim(line);
-    const auto fields = splitField(line, ' ');
-    DNSPacketWriter pw(packet, DNSName(fields.first), DNSRecordContent::TypeToNumber(fields.second));
+    if (line.empty() || line.at(0) == '#') {
+      continue;
+    }
+
+    auto fields = splitField(line, ' ');
```

Comment:
Would using `vector<string> parts; stringtok(parts, line, " ")` not be easier? Using the resulting vector's length to determine if subnet info was required (and ignoring it if the switch is not used).

Reply:
Sure, `stringtok()` is a bit more complex but that might makes things easier to read, so why not.